flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/StateChangelogWriter.java

```diff
@@ -73,12 +80,19 @@
      */
     void reset(SequenceNumber from, SequenceNumber to);
 
+    /**
+     * Truncate the tail of log and close it. No new appends will be possible. Any appended but not
+     * persisted records will be lost.
+     *
+     * @param from {@link SequenceNumber} from which to truncate the changelog, inclusive
+     */
+    default void truncateAndClose(SequenceNumber from) {
```

Comment:
The `StateChangelogWriter` class is an internal class, do we still need to introduce a default interface here?